tool/gravity/cli/clusterupdate.go

```diff
@@ -322,11 +321,26 @@ func checkForUpdate(
 		return nil, trace.Wrap(err)
 	}
 
-	apps, err := env.AppService(
-		defaults.GravityServiceURL,
-		localenv.AppConfig{},
-		httplib.WithLocalResolver(env.DNS.Addr()),
-		httplib.WithInsecure())
+	clusterEnv, err := localenv.NewClusterEnvironment()
+	if err != nil {
+		return nil, trace.Wrap(err)
+	}
+
+	clusterPackages, err := env.ClusterPackages()
+	if err != nil {
+		return nil, trace.Wrap(err)
+	}
+
+	// Create the *local* app service that uses the cluster's backend and
+	// packages.
+	//
+	// The local service is needed to handle cases such as newly introduced
+	// manifest field which gravity-site which may be running the old code
+	// does not recognize.
+	apps, err := env.AppServiceLocal(localenv.AppConfig{
```

Comment:
I'm now wondering why this is not part of the cluster environment - why not have `ClusterApps` (similar to `ClusterPackages` that uses the cluster application service)?

Reply:
Well... this app service is kind of a weird beast cause it uses cluster's packages/backend, but by itself is local... I guess I could put it in there anyway.